### ISSUE TITLE
Problematic Implementation of EMD Loss

### ISSUE THREAD

**Issue by VoVAllen** [kentsyx/Neural-IMage-Assessment#4]
Should be the L2 distance between CDF of two distributions but not between the PDF of two distributions

And there's some typo in the naming such as emb and emd

**Comment by VoVAllen** [comment]
Sorry for the mistake. I realized it's right. Thanks!
Another minor problem I found is Dropout. Due to pytorch features, you need to explicite call model.eval() in valuation and test period to turn the Dropout into right mode to get right forward result.

**Comment by kentsyx** [comment]
Fixed the dropout issue
I didn't see any typo that you mentioned

**Comment by VoVAllen** [comment]
可能是我看错了……应该没有typo……

**Comment by luqiang360** [comment]
https://github.com/kentsyx/Neural-IMage-Assessment/blob/9c50b3e384a88a8afdc00333d01656be5526bfed/model.py#L36

The EMD loss still seems to be wrong, my opinion is the sum operation should be inside of torch.abs
